silverstripe/silverstripe-admin
# FIX Add one to current colspan in fixColumns instead of resetting to 2

## Description
In the old situation, the final column in each row would be hardcoded to colspan="2" if the function 'fixColumns' was called. This is required because this function adds an additional column to the table that needs to be filled in. This change removes the hardcoded number and instead adds one to the current colspan of the last cell. This fixes the bug that the 'No items found' row would be capped at two columns even if the table has more.

## Manual testing steps
1. Manually create a GridField for an has_many relation that contains no records, this should fullfill the check in needsColumnFix by default. The relation class should have at least 3 summary fields.
2. Without this change, the 'No items found' row gets a colspan of 2. With this change, it should be 4.

## Issues

- #1859 

## Pull request checklist

    - See [picking the right version](https://docs.silverstripe.org/en/contributing/code/#picking-the-right-version)
    - Small amounts of additional linting are usually okay, but if it makes it hard to concentrate on the relevant changes, ask for the unrelated changes to be reverted, and submitted as a separate PR.
